src/core_ocean/mode_init/mpas_ocn_init_global_ocean.F

```diff
@@ -998,37 +994,35 @@ subroutine ocn_init_setup_global_ocean_create_model_topo(domain, iErr)!{{{
              call ocn_init_setup_global_ocean_deepen_critical_passages(meshPool, criticalPassagesPool, iErr)
           end if
 
-          ! Fill holes. Enforce that no cell has maxLevelCell that is more than
-          ! 1 vertical level higher than its neighbors.
-          if (config_global_ocean_fill_bathymetry_holes) then
-             call mpas_pool_get_subpool(block_ptr % structs, 'scratch', scratchPool)
-
-             call mpas_pool_get_field(scratchPool, 'smoothedLevels', smoothedLevelsField)
+          ! Alter cells for partial bottom cells before filling in bathymetry
+          ! holes.
+          do iCell = 1, nCells
+             call ocn_alter_bottomDepth_for_pbcs(bottomDepth(iCell), refBottomDepth, maxLevelCell(iCell), iErr)
+          end do
 
+          ! Fill bathymetry holes, i.e. single cells that are deeper than all neighbors.
+          ! These cells can collect tracer extrema and are not able to use
+          ! horizontal diffusion or advection to clear them out. Reduce pits to
+          ! make them level with the next deepest neighbor cell.
+          if (config_global_ocean_fill_bathymetry_holes) then
              call mpas_pool_get_array(meshPool, 'nEdgesOnCell', nEdgesOnCell)
              call mpas_pool_get_array(meshPool, 'cellsOnCell', cellsOnCell)
 
-             call mpas_allocate_scratch_field(smoothedLevelsField, .true.)
-
```

Comment:
I think I agree that the temporary field isn't needed but want to talk it through.  If a cell is deeper than its neighbors, it will get filled in.  By definition, it is now at the same maxLevelCell as its deepest neighbor.  Therefore, no neighbor can be deeper than this cell if it was altered and no neighbors will be filled.  So it should be safe to update maxLevelCell as we go.

Reply:
Yeah, I went through the same logic, and came to the same conclusion.